### PR TITLE
Add linting rule

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
## Why this PR?

Have a centralized copy of the .editorconfig file. I will also write a onenote page for it. But since our open-source client libraries also follow the linting rule set. I think here is a good place for it.

### CODE REVIEW

**Comment by zhenlan** [comment]
Who is the audience for these files? How are these files going to be used?

**Comment by zhiyuanliang-ms** [comment]
> Who is the audience for these files? How are these files going to be used?

App Config team and anyone wants to contribute to our open-source client libraries would be the audience.

I think we should have a centralized copy of the linting rule set (.editorconfig file) to be referenced and ensure every repo will use the same rule set. I do create an onenote page for it. But we also have feature management and .net provider which are on github. So I think this public repo is a better place to keep the linting rule set.

**Comment by juniwang** [comment]
Keeping a centralized copy as single source of truth would be a good idea. At the same time, it also brings in some sort of cross repo dependencies making it less convenient in setting up CI/CD(especially on ADO) and local env.

File: docs/Lint/charp/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# editorconfig.org
```
Comment (rossgrambo) [inline docs/Lint/charp/.editorconfig:1]: path should be `csharp`* instead of `charp`

**Comment by rossgrambo** [comment]
I like to think this is the .editorconfig for our team that we agreed on. Each repo might have slight variations (if there's reason to)- but this is the team's / would be used for a new repo.

Although I don't think I personally prefer for it to be public. It's not necessarily bad- but I don't see clear value.

**Comment by rossgrambo** [comment]
If we believe all `.editorconfig` files for our repos need to be tightly coupled / the same- I don't see value in having a copy here. I don't believe that's the case.

**Comment by zhenlan** [comment]
I'm with @rossgrambo on this. This repo is for customers, not centralized storage for our team. External developers who may contribute to our components will follow/obey the linting rules of respective repos.

File: docs/Lint/charp/.editorconfig (hunk 1 above)
Comment (zhiyuanliang-ms) [reply docs/Lint/charp/.editorconfig:1]: Good catch! My bad. Fixed.

**Comment by zhiyuanliang-ms** [comment]
@zhenlan 

> This repo is for customers

I agree with this. Then I will move the rule set copy to our onenote.

**Pull request closed without merging**